src/animation/PropertyBinding.js

```diff
@@ -102,47 +102,81 @@ Object.assign( PropertyBinding, {
 
 	},
 
-	parseTrackName: function ( trackName ) {
+	parseTrackName: function () {
 
-		// matches strings in the form of:
-		//    nodeName.property
-		//    nodeName.property[accessor]
-		//    nodeName.material.property[accessor]
-		//    uuid.property[accessor]
-		//    uuid.objectName[objectIndex].propertyName[propertyIndex]
-		//    parentName/nodeName.property
-		//    parentName/parentName/nodeName.property[index]
-		//    .bone[Armature.DEF_cog].position
-		//    scene:helium_balloon_model:helium_balloon_model.position
-		// created and tested via https://regex101.com/#javascript
+		// Parent directories, delimited by '/' or ':'. Currently unused, but must
+		// be matched to parse the rest of the track name.
+		var directoryRe = /((?:[\w-]+[\/:])*)/;
 
-		var re = /^((?:[\w-]+[\/:])*)([\w-]+)?(?:\.([\w-]+)(?:\[(.+)\])?)?\.([\w-]+)(?:\[(.+)\])?$/;
-		var matches = re.exec( trackName );
+		// Target node. May contain word characters (a-zA-Z0-9_) and '.' or '-'
+		// characters, but must begin and end with a word character.
+		var nodeRe = /(\w(?:[\w-\.]*\w)?)?/;
 
-		if ( ! matches ) {
+		// Object on target node, and accessor. May contain only word characters,
+		// and must be a member of the supportedObjectNames whitelist. Accessor may
+		// contain any non-bracket characters.
+		var objectRe = /(?:\.([\w-]+)(?:\[(.+)\])?)?/;
```

Comment:
I meant, it'd be helpful to write here the result is compared with the whilelist afterwards and why.

When I first read this comment, I thought the whilelist should be in the regex because
it mentions `must be a member of the supportedObjectNames whitelist`.